src/types.rs

```diff
@@ -349,30 +347,12 @@ pub enum ReadStreamStatus<A> {
     Error(ReadStreamError),
 }
 
-pub enum Payload {
-    Json(Bytes),
-    Binary(Bytes),
-}
-
-impl Payload {
-    pub fn is_json(&self) -> bool {
-        matches!(*self, Payload::Json(_))
-    }
-
-    pub fn into_inner(self) -> Bytes {
-        match self {
-            Payload::Json(bytes) => bytes,
-            Payload::Binary(bytes) => bytes,
-        }
-    }
-}
-
 /// Holds data of event about to be sent to the server.
 pub struct EventData {
-    pub(crate) event_type: String,
-    pub(crate) payload: Payload,
+    pub(crate) payload: Bytes,
```

Comment:
@YoEight sidenote, we should probably rename it to `data` as we have in other clients.

Reply:
Yes, let's do that on a different PR.